test/common/protobuf/utility_test.cc

```diff
@@ -1700,78 +1700,203 @@ TEST_F(ProtobufUtilityTest, GetYamlStringFromProtoInvalidAny) {
 }
 
 TEST(DurationUtilTest, OutOfRange) {
-  {
-    ProtobufWkt::Duration duration;
-    duration.set_seconds(-1);
-    EXPECT_THROW(DurationUtil::durationToMilliseconds(duration), EnvoyException);
-  }
-  {
-    ProtobufWkt::Duration duration;
-    duration.set_nanos(-1);
-    EXPECT_THROW(DurationUtil::durationToMilliseconds(duration), EnvoyException);
-  }
-  {
-    ProtobufWkt::Duration duration;
-    duration.set_nanos(1000000000);
-    EXPECT_THROW(DurationUtil::durationToMilliseconds(duration), EnvoyException);
-  }
-  {
-    ProtobufWkt::Duration duration;
-    duration.set_seconds(Protobuf::util::TimeUtil::kDurationMaxSeconds + 1);
-    EXPECT_THROW(DurationUtil::durationToMilliseconds(duration), EnvoyException);
-  }
-  {
-    ProtobufWkt::Duration duration;
-    constexpr int64_t kMaxInt64Nanoseconds =
-        std::numeric_limits<int64_t>::max() / (1000 * 1000 * 1000);
-    duration.set_seconds(kMaxInt64Nanoseconds + 1);
-    EXPECT_THROW(DurationUtil::durationToMilliseconds(duration), EnvoyException);
+  // Once the runtime feature "envoy.reloadable_features.strict_duration_validation"
+  // is deprecated, this test should only validate the "true" case.
+  for (const std::string strict_duration : {"true", "false"}) {
+    TestScopedRuntime scoped_runtime;
+    scoped_runtime.mergeValues(
+        {{"envoy.reloadable_features.strict_duration_validation", strict_duration}});
+    {
+      ENVOY_LOG_MISC(trace, "ADIP: here1");
+      ProtobufWkt::Duration duration;
+      duration.set_seconds(-1);
+      EXPECT_THROW(DurationUtil::durationToMilliseconds(duration), EnvoyException);
+    }
+    {
+      ENVOY_LOG_MISC(trace, "ADIP: here2");
+      ProtobufWkt::Duration duration;
+      duration.set_nanos(-1);
+      EXPECT_THROW(DurationUtil::durationToMilliseconds(duration), EnvoyException);
+    }
+    // Invalid number of nanoseconds.
+    {
+      ENVOY_LOG_MISC(trace, "ADIP: here3");
+      ProtobufWkt::Duration duration;
+      duration.set_nanos(1000000000);
+      EXPECT_THROW(DurationUtil::durationToMilliseconds(duration), EnvoyException);
+    }
+    {
+      ENVOY_LOG_MISC(trace, "ADIP: here4");
+      ProtobufWkt::Duration duration;
+      duration.set_seconds(Protobuf::util::TimeUtil::kDurationMaxSeconds + 1);
+      EXPECT_THROW(DurationUtil::durationToMilliseconds(duration), EnvoyException);
+    }
+    // Invalid number of seconds.
+    {
+      ENVOY_LOG_MISC(trace, "ADIP: here5");
```

Comment:
remove these before merging

Reply:
Oops... thanks for catching this.
Done.